### DISCUSSION TITLE
Directus Flow - Request URL -> Create Item in collection

### DISCUSSION THREAD

**Discussion by ThomasBonnelye** [directus/directus#15439] [Archive]
Hello !

I am discovering the new "Flow" system on directus and I am facing a problem that I can't find a solution to on the Github / Doc.

I'm trying to get data from the Steam api from this url : [(http://api.steampowered.com/ISteamNews/GetNewsForApp/v0002/?appid=570&count=3&maxlength=0&format=json&feeds=steam_community_announcements)]

To then create directly my items in a collection.

**But I have two problems:**

- How to define the paylod of my item creation in order to be dependent on the retrieved data?
I currently have this:
`[
    {
        "gid": "{{$request_t7urr.data.appnews.newsitems.gid}}",
        "title": "article-1",
        "status": "published",
        "content": "An interesting article about web"
    }
]`

I can't find anything in the documentation explaining how to link the previously retrieved data to this new operation.

- Second question, how to define a filter on the same data to avoid creating the same item twice?

I also add some pictures to illustrate what I am saying

![data-model](https://user-images.githubusercontent.com/32672029/188737644-cf07e916-255d-4cb0-9dec-0ef12ada804c.PNG)
![create-item](https://user-images.githubusercontent.com/32672029/188737645-0d0444ac-aed9-4896-8c5e-cebd7ebc3fce.PNG)
![webhook-request](https://user-images.githubusercontent.com/32672029/188737646-24e8e39d-b5c0-4c85-a18e-3bb0bd078e2f.PNG)
![flow-schema](https://user-images.githubusercontent.com/32672029/188737648-3c1958b5-eceb-4de7-9da2-8e6ddb59b8af.PNG)

Thanks in advance !

**Comment by itsmatteomanf** [comment]
Hello!

Going in reverse order.
For Q2 you have to read the data based on the ID you use, check if it's empty and then decide there.
For Q1 you should check what you are exactly getting from the logs or a specific console log action, I have seen not very consistent data structures. For sure previous actions don't have a `$` in front of the ID. You could also use `$last` to access the last operation.

**Reply by ThomasBonnelye** [reply]
Good evening! Thank you for your answer!

The data model I get is the following:
`"data": {
    "appnews": {
      "appid": 570,
      "count": 589,
      "newsitems": [
        {
          "gid": "4621277624691973561",
          "url": "https://steamstore-a.akamaihd.net/news/externalpost/steam_community_announcements/4621277624691973561",
          "date": 1662064864,
          "tags": [
            "mod_reviewed",
            "ModAct_444533432_1662066719_0",
            "mod_require_rereview"
          ]...,`
          
So to access my data, I have to try something like `"gid" : "{{$last.data.appnews.newsitems.gid}}"?`

**Reply by itsmatteomanf** [reply]
Yeah, it seems to be the correct behaviour. My suggestion is to have a console block to test if it works without anything else going on.

**Comment by ThomasBonnelye** [comment]
Thanks for your answer and advice, I'm testing this tomorrow!

Le mer. 7 sept. 2022 à 00:59, Matteo Manfredi ***@***.***> a
écrit :

**Reply by ThomasBonnelye** [reply]
Hello ! 
This was the right solution, but I still have one last problem to which I cannot find a solution.
I get an array in the output of my url request and I can't loop over it to create an item for each element of the array.

`"gid": "{{$last.data.appnews.newsitems[0].gid}}", -> works (i get  the first element)
 "gid": "{{$last.data.appnews.newsitems[*].gid}}", -> not working`
 
 I tried to create a script as shown in this  thread : #15101  but I don't see how to trigger the creation of element for each element.

**Reply by itsmatteomanf** [reply]
From what I know loops aren't supported... not sure if you can add data via an array of objects, like you were importing a file. It should work in my mind, but have no clue if they actually implemented it that way.

**Comment by ThomasBonnelye** [comment]
Hello !

I finally chose a simpler solution, I push directly all the data in a JSON input and I process the data in the front of my application.

But I can't update my item, only create new ones.

I think I have filled in the update data block correctly. Do you have any idea why it doesn't work ? (In the logs no access problem, and the payload is the same as the one used for the item creation and it works).

By logic it must come from the "IDs" field but I don't really see what the problem is here.

(This technique is not the cleanest but I don't have enough time to process this with a clean loop.
I'll put my final flow here when I've taken the time to create it!)

 Here are some screenshots :
 
![image](https://user-images.githubusercontent.com/32672029/189335441-7f350bd9-558d-45eb-b9d2-0afb2deef061.png)
![image](https://user-images.githubusercontent.com/32672029/189335542-ba4497dd-9798-41a2-8ed6-67abafab3fba.png)
![image](https://user-images.githubusercontent.com/32672029/189335639-ac43ed91-0361-4bd2-8cd6-7c99c6237255.png)

**Reply by rijkvanzanten** [reply]
@ThomasBonnelye Hard to tell exactly with the limited info, but a first thing that comes to mind for me is: is the field called `Steam_data` or `steam_data`? The uppercase `S` there makes me go 🤨

**Reply by ThomasBonnelye** [reply]
Hello ! Thank you for your feedback!
The name of the field is good. The thing I don't understand is that the
creation of data with this Payload works, but not the update of a data
(even the status field does not change). So no error, just an empty Payload
in return: "{}".
The id is also the right one because Read data works on this item too.

What info / screenshot would you like to have?

**Reply by rijkvanzanten** [reply]
Does the current user that's triggering the flow have permissions to update that item? I noticed that the operation is configured to inherit the permissions from the thing triggering the flow 🤔

**Reply by ThomasBonnelye** [reply]
Yes the current user is admin and have all permissions !
